mautic/mautic
# Integer range point action validation

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ Yes]
| New feature/enhancement? (use the a.x branch)      | [No ]
| Deprecations?                          | [ No]
| BC breaks? (use the c.x branch)        | [ No]
| Automated tests included?              | [ Yes] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #...  



#### Description:

Saving a point action with an insanely large point value (positive or negative) shows a proper message now instead of a 500.



#### Steps to test this PR:



1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Go to Point action
3. Add New
4. Enter name
5. Select any When a contact... condition
6. Enter points as "0345738945739"
7. Click Save and close.
8. There should be an error saying "This value should be between -2147483648 and 2147483647."

[//]: # ( As applicable: )
#### List of areas covered by the unit and/or functional tests:
1. Validation for range on add and edit points.
2.  Test cases include the following cases: saving within range, saving with the upper limit value (2147483647), saving with lower limit value (-2147483648), saving with a value above upper limit and saving with a value below lower limit.
